src/include/interval_set.h

```diff
@@ -249,44 +249,60 @@ class interval_set {
  private:
   auto find_inc(T start) const {
     auto p = m.lower_bound(start);  // p->first >= start
-    if (p != m.begin() &&
-        (p == m.end() || p->first > start)) {
-      --p;   // might overlap?
-      if (p->first + p->second <= start)
-        ++p; // it doesn't.
+    while (p != m.begin()) {
+        if (p->first <= start) {
+	  break;
+        } else {
+	  --p;   // might overlap?
+        }
+    }
+    if (p->first + p->second <= start) {
+      ++p; // it doesn't.
     }
     return p;
   }
   
   auto find_inc_m(T start) {
     auto p = m.lower_bound(start);
```

Comment:
I am not sure why the existing code use `lower_bound()`, IMO we should switch to use the `upper_bound()` instead.

For example:

```
[40] = 10
[60] = 10
[100] = 30
```
If the `start` is `40` the `lower_bound()` will return `.end()`, right ? Then the `--p` will be `[100] = 30` ?

While the `upper_bound()` will return `[40] = 10`.

Reply:
Okay, you are right. I misread it. Thanks.